src/main/java/net/glowstone/block/blocktype/BlockPortal.java

```diff
@@ -0,0 +1,87 @@
+package net.glowstone.block.blocktype;
+
+import java.util.concurrent.ThreadLocalRandom;
+import net.glowstone.GlowServer;
+import net.glowstone.GlowWorld;
+import net.glowstone.block.GlowBlock;
+import net.glowstone.entity.monster.GlowPigZombie;
+import net.glowstone.entity.physics.BoundingBox;
+import net.glowstone.util.pattern.PortalShape;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.BlockFace;
+import org.bukkit.event.entity.CreatureSpawnEvent;
+import org.bukkit.util.Vector;
+
+public class BlockPortal extends BlockType {
+
+    @Override
+    public void onNearBlockChanged(final GlowBlock block, BlockFace face, GlowBlock changedBlock,
+                                   Material oldType, byte oldData, Material newType, byte newData) {
+        if (newType == oldType || (oldType != Material.PORTAL && oldType != Material.OBSIDIAN)) {
+            return;
+        }
+        BlockFace left = getFace(block.getData());
+        if (left == null) {
+            return;
+        }
+        PortalShape shape = new PortalShape(block.getLocation(), left);
+        if (shape.validate()
+                && shape.getPortalBlockCount() == shape.getHeight() * shape.getWidth()) {
+            return;
+        }
+        block.setType(Material.AIR);
+    }
+
+    public static BoundingBox getBoundingBox(GlowBlock block) {
```

Comment:
This needs documentation, specifying it only applies to portal blocks.

Reply:
Applied